nest-forms-backend/src/convert/convert.service.ts

```diff
@@ -373,43 +321,33 @@ export default class ConvertService {
     res: Response,
     user?: CognitoGetUserData,
   ): Promise<StreamableFile> {
-    const schemaVersion = await this.prismaService.schemaVersion.findUnique({
-      where: {
-        id: data.schemaVersionId,
-      },
-      include: {
-        schema: true,
-      },
-    })
-
-    if (schemaVersion === null) {
-      throw this.throwerErrorGuard.NotFoundException(
-        SchemasErrorsEnum.SCHEMA_VERSION_NOT_FOUND,
-        SchemasErrorsResponseEnum.SCHEMA_VERSION_NOT_FOUND,
-      )
-    }
-
     // for eligible tax forms (those of signed-in users) store json before transform and the transformed pdf itself for debug purposes
     let taxDebugBucket = ''
     let directoryName = ''
     let stringifiedData = ''
     let shouldStoreDebugPdfData = false
-    const taxFormPospId = process.env.TAX_FORM_POSP_ID
 
     const form = await this.formsService.getFormWithAccessCheck(
       data.formId,
       user?.sub ?? null,
       ico,
-      true,
     )
 
+    const formDefinition = getFormDefinitionBySlug(form.formDefinitionSlug)
+    if (!formDefinition) {
+      throw this.throwerErrorGuard.NotFoundException(
+        FormsErrorsEnum.FORM_DEFINITION_NOT_FOUND,
+        `${FormsErrorsResponseEnum.FORM_DEFINITION_NOT_FOUND} ${form.formDefinitionSlug}`,
+      )
+    }
+
     let formJsonData = data.jsonData
     if (formJsonData === undefined) {
       formJsonData = form.formDataJson
     }
 
     // common init for both json and pdf debug storage
-    if (taxFormPospId && schemaVersion.pospID === taxFormPospId && user) {
+    if (isSlovenskoSkTaxFormDefinition(formDefinition)) {
```

Comment:
Is missing `user` condition ok?

Reply:
I think so, if it is not used in the body.